script.js

```diff
@@ -1,47 +1,99 @@
-let currentInput = "";
-let memory = [];
-const display = document.getElementById('display');
-
-function appendToDisplay(value) {
-    currentInput += value;
-    display.value = currentInput;
-}
-
-function clearDisplay() {
-    currentInput = "";
-    display.value = "";
-}
-
-function calculateResult() {
-    try {
-        const result = eval(currentInput);
-        display.value = result;
-        currentInput = result.toString();
-        saveToMemory(currentInput, result);
-    } catch (error) {
-        display.value = "Error";
-    }
-}
-
-function storeInMemory() {
-    try {
-        const result = eval(currentInput);
-        saveToMemory(currentInput, result);
-    } catch (error) {
-        display.value = "Error";
-    }
-}
-
-function recallMemory() {
-    if (memory.length > 0) {
-        const lastCalculation = memory[memory.length - 1];
-        display.value = `${lastCalculation.input} = ${lastCalculation.result}`;
-        currentInput = lastCalculation.input;
-    }
-}
-
-function saveToMemory(input, result) {
-    memory.push({ input, result });
-}
+$(document).ready(function() {
+    // We write all jQuery related code inside this function
+    let currentInput = "";
+    let memory = [];
+    let memoryIndex = null;
+    let display = $('#displayField');
+    
+    // This is how you execute a function when a html element with the id=addMemory is clicked
+    $('#addMemory').on('click', () => {
+        // .val() is the value of the input field
+        var calculation = display.val();
+        memory.push(calculation);
+        memoryIndex = memory.length - 1;
+        // Set the display field to the result of the calculation
+        display.val(eval(calculation));
+    });
+
+    //Note: () => {} is same as function() {}
+
+    $('#getNextCalculation').on('click', () => {
+        var memoryIndex = memoryIndex;
```

Comment:
This is where it goes wrong. Why do you save the value of global memoryIndex variable to a local memoryIndex..?